sktime/annotation/igts.py

```diff
@@ -0,0 +1,486 @@
+                ChangePointResult(
+                    k=k, score=ig_max, change_points=current_change_points.copy()
+                )
+            )
+
+            # logger.info(
+            #     f"BEST {ig_max=:.5f} for {best_candidate=} "
+            #     + f"current {current_change_points=}"
+            # )
+        return current_change_points
+
+
+class SegmentationMixin:
+    """Mixin with methods useful for segmentation problems."""
+
+    def to_classification(self, change_points: List[int]) -> npt.ArrayLike:
+        """Convert change point locations to a classification vector.
+
+        Change point detection results can be treated as classification
+        with true change point locations marked with 1's at position of
+        the change point and remaining non-change point locations being
+        0's.
+
+        For example change points [2, 8] for a time series of length 10
+        would result in: [0, 0, 1, 0, 0, 0, 0, 0, 1, 0].
+        """
+        return np.bincount(change_points[1:-1], minlength=change_points[-1])
+
+    def to_clusters(self, change_points: List[int]) -> npt.ArrayLike:
+        """Convert change point locations to a clustering vector.
+
+        Change point detection results can be treated as clustering
+        with each segment separated by change points assigned a
+        distinct dummy label.
+
+        For example change points [2, 8] for a time series of length 10
+        would result in: [0, 0, 1, 1, 1, 1, 1, 1, 2, 2].
+        """
+        labels = np.zeros(change_points[-1], dtype=np.int32)
+        for i, (start, stop) in enumerate(zip(change_points[:-1], change_points[1:])):
+            labels[start:stop] = i
+        return labels
+
+
+class InformationGainSegmentation(SegmentationMixin, BaseEstimator):
+    """Information Gain based Temporal Segmentation (IGTS) Estimator.
+
+    IGTS is a n unsupervised method for segmenting multivariate time series
+    into non-overlapping segments by locating change points that for which
+    the information gain is maximized.
+
+    Information gain (IG) is defined as the amount of entropy lost by the segmentation.
+    The aim is to find the segmentation that have the maximum information
+    gain for a specified number of segments.
+
+    IGTS uses top-down search method to greedily find the next change point
+    location that creates the maximum information gain. Once this is found, it
+    repeats the process until it finds `k_max` splits of the time series.
+
+    .. note::
+
+       IGTS does not work very well for univariate series but it can still be
+       used if the original univariate series are augmented by an extra feature
+       dimensions. A technique proposed in the paper [1]_ us to subtract the
+       series from it's largest element and append to the series.
+
+    Parameters
+    ----------
+    k_max: int, default=10
+        Maximum number of change points to find. The number of segments is thus k+1.
+
```

Comment:
```suggestion
```

Reply:
I think it's more readable with a line break